app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -503,19 +526,14 @@ private fun AssociationDescription(association: Association) {
  *
  * !!! The follow button do not have any action and should be implemented when follow feature is
  * implemented !!!
- *
- * @param association (Association) : The association currently displayed
- * @param context (Context) : The context of the screen
  */
 @Composable
 private fun AssociationHeader(
-    associationViewModel: AssociationViewModel,
-    userViewModel: UserViewModel,
-    user: User
+    association: Association,
+    isFollowed: Boolean,
+    enableButton: Boolean,
+    onFollow: () -> Unit,
 ) {
-  val association = associationViewModel.selectedAssociation.collectAsState().value!!
-  var isFollowed by remember { mutableStateOf(user.isFollowAssociation(association)) }
-  var isButtonEnabled by remember { mutableStateOf(true) }
```

Comment:
Very good implementation to remove the view models ! We should now pay attention that the argument given to the @Composable like this are collectAsState thought, not to loose the automatic refresh ; )